dbms/src/Flash/tests/gtest_collation.cpp

```diff
@@ -0,0 +1,436 @@
+    using ColStringNullableType = std::optional<typename TypeTraits<String>::FieldType>;
+    using ColStringType = std::optional<typename TypeTraits<String>::FieldType>;
+    using ColUInt64Type = typename TypeTraits<UInt64>::FieldType;
+
+    using ColumnWithNullableString = std::vector<ColStringNullableType>;
+    using ColumnWithString = std::vector<ColStringType>;
+    using ColumnWithUInt64 = std::vector<ColUInt64Type>;
+
+    void initializeContext() override
+    {
+        ExecutorTest::initializeContext();
+
+        context.addMockTable({db_name, table_name},
+                             {{col_name, TiDB::TP::TypeString}},
+                             {toNullableVec<String>(col_name, col)});
+
+        context.addMockTable({db_name, chinese_table},
+                             {{chinese_col_name, TiDB::TP::TypeString}},
+                             {toNullableVec<String>(chinese_col_name, chinese_col)});
+
+        context.addMockTable(join_table, "t1", {{"a", TiDB::TP::TypeLong}, {"b", TiDB::TP::TypeLong}}, {toVec<Int32>("a", {1, 1, 3, 4}), toVec<Int32>("b", {1, 1, 4, 1})});
+
+        context.addMockTable(join_table, "t2", {{"a", TiDB::TP::TypeLong}, {"b", TiDB::TP::TypeLong}}, {toVec<Int32>("a", {1, 4, 2}), toVec<Int32>("b", {2, 6, 2})});
+
+        /// For topn
+        context.addMockTable({db_name, topn_table},
+                             {{topn_col, TiDB::TP::TypeString}},
+                             {toNullableVec<String>("_col", ColumnWithString{"col0-0", "col0-1", "col0-2", {}, "col0-4", {}, "col0-6", "col0-7"})});
+
+        /// For projection
+        context.addMockTable({db_name, proj_table},
+                             {{proj_col[0], TiDB::TP::TypeString},
+                              {proj_col[1], TiDB::TP::TypeString}},
+                             {toNullableVec<String>(proj_col[0], ColumnWithString{"col0-0", "col0-1", "", "col0-2", {}, "col0-3", ""}),
+                              toNullableVec<String>(proj_col[1], ColumnWithString{"", "col1-1", "", "col1-0", {}, "col1-3", "col1-2"})});
+
+        /// For limit
+        context.addMockTable({db_name, limit_table},
+                             {{limit_col, TiDB::TP::TypeString}},
+                             {toNullableVec<String>(limit_col, ColumnWithString{"col0-0", {}, "col0-2", "col0-3", {}, "col0-5", "col0-6", "col0-7"})});
+
+        /// For ExchangeSender
+        context.addExchangeRelationSchema(sender_name, {{"s1", TiDB::TP::TypeString}, {"s2", TiDB::TP::TypeString}, {"s3", TiDB::TP::TypeString}});
+    }
+
+    void setAndCheck(const String & table_name, const String & col_name, Int32 collation, const ColumnsWithTypeAndName & expect)
+    {
+        context.setCollation(collation);
+        auto request = context.scan(db_name, table_name).aggregation(MockAstVec{}, {col(col_name)}).project({col_name}).build(context);
+        std::cout << request->DebugString() << std::endl;
+        ASSERT_COLUMNS_EQ_UR(expect, executeStreams(request, 1));
```

Comment:
seems `executeStreams(reuqest)` is enough.

Reply:
> seems `executeStreams(reuqest)` is enough.

ok